doc: update known issues for NCS 2.3.0 (#10362)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

   When adding a new version, add it to the dropdown list above and move the "selected" option next to it.
   Once "selected" is moved, only issues that are valid for the new version will be displayed when entering the page.

   Known issues process is described at https://projecttools.nordicsemi.no/confluence/pages/viewpage.action?pageId=82556815

   When updating this file, add entries in the following format:

   .. rst-class:: vXXX vYYY

>    JIRA-XXXX: Title of the issue (with mandatory JIRA issue number since NCS v1.7.0)

Reviewer: ```suggestion
   JIRA-XXXX: Title of the issue (with mandatory JIRA issue number since nRF Connect SDK v1.7.0)
```

Asset Tracker v2
================

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 2: ```suggestion
.. rst-class:: v2-2-0
```

No longer getting the build error. I also tried on 2.2.0 and it succeeded which is weird.
@gregersrygg Could you try to build with `overlay-carrier.conf` on the affecting versions?

Reviewer 3: I could build atv2 for thingy:91 with `overlay-carrier.conf` from `main`, but `v2.2-branch` failed for me with the same error as before:

    Error: Image size (0x68a34) + trailer (0x630) exceeds requested size 0x69000

Reviewer 2: Alright, thanks! @greg-fe then the suggestion is valid. Remove the entry for 2.3.0.

CIA-890: The application cannot be built with :file:`overlay-debug.conf` and :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``
  Due to insufficient flash space for the application when it is not optimized, the :ref:`asset_tracker_v2` application cannot be built with :file:`overlay-debug.conf` and :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``.

>   **Workaround:** To free up flash space when debugging locally, uncomment the following Kconfig options in the :file:`prj.conf` file:

Reviewer 3: ```suggestion
  **Workaround:** To free up flash space when debugging locally, comment out the following Kconfig options in the :file:`prj.conf` file:
```

Other issues
============

.. rst-class:: v2-3-0 v2-2-0

CIA-857: LTE Link Controller spurious events
  When using the :ref:`lte_lc_readme` library, a memory comparison is done between padded structs that may result in comparing bytes that have undefined initialization values.
  This may cause spurious ``LTE_LC_EVT_CELL_UPDATE`` and ``LTE_LC_EVT_PSM_UPDATE`` events even though the information contained in the event has not changed since last update event.

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Reviewer 4: Issue fixed for v2.3.0 (Merged here https://github.com/nrfconnect/sdk-nrf/pull/8960)

> Other issues

Reviewer 5: As part of zephyr upmerge, CONFIG_LOG_PRINTK was change from default 'n' to 'y'. This may cause issue in applications which have used printk(). Together with a bug (CIA-892) when logging strings of more than 1023 characters, this may introduce new crashes/asserts.
We are still debating the resolution to CIA-892 for the NCS 2.3.0 release so not sure if this should be a known issue but I'll record it here so we don't forget to mention it.
Eventually, https://github.com/zephyrproject-rtos/zephyr/pull/54901 will resolve this.

Reviewer 5: It seems there's been a mitigating hack for CIA-892 for asset tracker in PR https://github.com/nrfconnect/sdk-nrf/pull/10458 but it seems the real fix is not taken for 2.3.0. If that's the case, we need a statement either in known issues or probably preferably in release notes that `CONFIG_LOG_PRINTK` has been changed from 'n' to 'y'. Below is some example of what we could write:

"When logging a string with %s qualifier, the maximum length of any inserted string is 1023 bytes, or otherwise an assert or a crash will occur. Default value of `CONFIG_LOG_PRINTK` Kconfig option has been changed from 'n' to 'y' in this release so even use of printk() function may cause this issue unless the application disables this option."

Other issues
============

.. rst-class:: v2-3-0

> CIA-892: Assert when printing strings with the ``%s`` qualifier

Reviewer 5: This is assert or crash, I think. In ATv2 we saw a crash. I've seen an assert also.

Reviewer 6: Correct, in ATv2 we saw crashes when using `printk()` to print large strings. handled in #10458 for ATv2, but the same may happen in other applications

> CIA-892: Assert when printing strings with the ``%s`` qualifier

Reviewer 5: "printing strings" -> "printing long strings"

  When logging a string with the ``%s`` qualifier, the maximum length of any inserted string is 1023 bytes.

>   If the string is longer, an assert or a crash will occur.

Reviewer 6: It's not certain that a crash will happen. The stack is corrupted, so the subsequent behavior is undefined

Author: Changed "will" to "can".

  The time returned by :ref:`lib_date_time` library becomes incorrect after one week elapses.
  This is due to an issue with clock_gettime() API.

.. rst-class:: v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0

TNSW-46156: The LwM2M carrier library can in some cases trigger a modem bug related to PDN deactivation (AT+CGACT) in NB-IoT mode.

>   This causes excessive network signaling which can make interactions with the modem (AT commands or :ref:`nrfxlib:nrf_modem` library functions) hang for several minutes.

Reviewer: nrf_modem already expands to Modem library.
```suggestion
  This causes excessive network signaling which can make interactions with the modem (AT commands or :ref:`nrfxlib:nrf_modem` functions) hang for several minutes.
```

Reviewer: Needs to be fixed

    west build --board nrf9160dk_nrf9160_ns@1.1.0

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

Reviewer 4: Issue fixed for v2.3.0 (merged here https://github.com/nrfconnect/sdk-nrf/pull/9873)

CIA-738: FMFU does not use external flash partitions
  Full modem FOTA support assumes full control of the external flash; it does not use an external flash partition.
  It cannot be combined with other usages, such as storing settings or P-GPS data.

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 4: Issue fixed for v2.3.0 (merged here https://github.com/nrfconnect/sdk-nrf/pull/9540)

nRF5340
=======

.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

> NCSDK-16856: Increased power consumption observed for the Low Power UART sample on nRF5340 DK

Reviewer 7: Fixed in 2.3 (see #9606).

  When using external flash with P-GPS, stored predictions cannot be reliably read to satisfy the modem's need for ephemeris data when actively downloading and storing predictions to the same external flash.
  Once the download is complete, the predictions can be reliably read.

> nRF7002

Reviewer 8: Please add nRF7 as the main heading and nRf7002 under that?

Author: There is no other nRF7 SoC that would justify this. For now, I'll use the approach for nRF9160. If there are more SoCs from the 91 family or the nrf7 family, I'll add the respective headers.

Reviewer 8: OK

*******

.. rst-class:: v2-3-0

SHEL-1352: Incorrect base address used in the OTP TX trim coefficients

>   Because of the use of an incorrect base address used in the One-Time Password TX trim coefficients, the transmit power will deviate by +/- 2dB from the target value.

Author: @bama-nordic , please review.

Reviewer 8: I suggest rewording the description as "Incorrect base address used for TX trim coefficients in the One-Time Programmable (OTP) memory, resulting in transmit power deviating by +/- 2dB from the target value"

Nordic Thingy:53
================

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 9: Already fixed

Matter
======

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1

Reviewer 10: @greg-fer regarding all three known-issues related to Android CHIPTool. Should we update the version number, if Android CHIPTool was deprecated and we do not recommend using it anymore?

Author: Added the following note:

  .. note::
      The support for the Android CHIP Tool is removed as of the |NCS| v2.3.0 for Matter in the |NCS|. Use CHIP Tool for Linux or macOS instead, as described in :ref:`ug_matter_gs_testing`.

KRKNWK-15846: Android CHIP Tool crashes when subscribing in the :guilabel:`LIGHT ON/OFF & LEVEL CLUSTER`
  The Android CHIP Tool crashes when attempting to start the subscription after typing minimum and maximum subscription interval values.
  Also, the Subscription window in the :guilabel:`LIGHT ON/OFF & LEVEL CLUSTER` contains faulty GUI layout (overlapping captions) used when passing minimum and maximum subscription interval values.
  This affects the Android CHIP Tool revision used for the |NCS| v2.2.0, v2.1.1, and v2.1.2 releases.

> .. rst-class:: v2-3-0 v2-2-0 v2-1-2 v2-1-1

Reviewer 10: It was fixed and should not be present in 2.3.0 anymore.

nRF Desktop
===========

.. note::
    nRF Desktop is also affected by the Bluetooth LE issue :ref:`NCSDK-19865 <ncsdk_19865>`.

.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0

NCSDK-8304: HID configurator issues for peripherals connected over Bluetooth® LE to Linux host
  Using :ref:`nrf_desktop_config_channel_script` for peripherals connected to host directly over Bluetooth LE might result in receiving improper HID feature report ID.
  In such case, the device will provide HID input reports, but it cannot be configured with the HID configurator.

  **Workaround:** Use BlueZ in version 5.56 or higher.

.. rst-class:: v2-2-0

> NCSDK-19970: MCUboot bootloader fails to swap images on nRF52840 DK that uses external FLASH

Reviewer: ```suggestion
NCSDK-19970: MCUboot bootloader fails to swap images on nRF52840 DK that uses external Flash
```

   The MCUboot bootloader cannot access external flash because the chosen ``nordic,pm-ext-flash`` is not defined in the MCUboot child image's devicetree.
   As a result ``nrf52840dk_nrf52840`` using ``prj_mcuboot_qspi.conf`` configuration fails to swap images after a complete image transfer.

   **Workaround**: Manually cherry-pick and apply the commit with the fix from the ``main`` branch (commit hash: ``7cea1b7e681a39ce2e2143b6b03132d95b7606ab``).
   Make sure to also cherry-pick and apply the commit that fixes a build system issue (commit `ec23df1fa305e99194ceac87a028f6da206a3ff1` from ``main`` branch).

>    This is needed to ensure that the introduced DTS overlay will be applied to the mcuboot child image.

Reviewer: ```suggestion
   This is needed to ensure that the introduced DTS overlay will be applied to the MCUboot child image.
```

nRF5340 Audio
=============

.. rst-class:: v2-3-0 v2-2-0

OCT-2347: Stream reestablishment issues in CIS

>   In the CIS mode, if a stream is running and the headset is reset, the gateway cannot reestablish the stream properly.

Reviewer 11: No longer an issue

> .. rst-class:: v2-3-0 v2-2-0
> 
> OCT-2347: Stream reestablishment issues in CIS
>   In the CIS mode, if a stream is running and the headset is reset, the gateway cannot reestablish the stream properly.
> 

Reviewer 11: ```suggestion

```

Reviewer 11: This is fixed

> .. rst-class:: v2-3-0 v2-2-0
> 
> OCT-2347: Stream reestablishment issues in CIS
>   In the CIS mode, if a stream is running and the headset is reset, the gateway cannot reestablish the stream properly.
> 

Reviewer 11: ```suggestion
```
This has been fixed

nRF5340 Audio
=============

.. rst-class:: v2-3-0

OCT-2470: Discovery of MCS failed
  When restarting or resetting the gateway, one or more headsets may run into a condition where the discovery of the Media Control Service fails.

>   This is related to the Bluetooth LE Audio Service.

Reviewer 12: This line should be removed.

nRF Machine Learning
====================

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 9: btw. Could we also slightly improve this known issue?

```
   The MCUboot bootloader cannot access external flash because the chosen ``nordic,pm-ext-flash`` is not defined in MCUboot child image's devicetree.
   As a result, MCUboot cannot swap images that are received during the OTA update.

**Workaround**: Manually cherry-pick and apply the commit with the fix from the ``main`` branch (commit hash: ``f54f6bbd423b12a595e76425e688f034926b8018``) to fix the issue for nrf5340dk_nrf5340_cpuapp. The same fix needs to be applied for the Thingy:53 board.
```

.. _ncsidb_925:

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

Reviewer 9: Should be fixed in NCS 2.3.
@rakons, could you double check, please ?

Reviewer 13: Yes. Fixed by https://github.com/nrfconnect/sdk-nrf/pull/10093

Bluetooth® LE
=============

> 

Reviewer 14: Please add the following new issue (we haven't got a workaround yet):
.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0

NCSDK-19942: HID samples don't work with Android 13
  The peripheral_hids_keyboard and peripheral_hids_mouse enter a connection-disconnection loop when trying to connect to them from Android 13.

Bluetooth® LE
=============

.. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0
.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0

> NCSDK-19727: Cannot build the :ref:`peripheral_hids_mouse` sample with security disabled.

Reviewer 14: Could you please leave this issue (not remove it)? It has been added together with a bugifx PR; https://github.com/nrfconnect/sdk-nrf/pull/10251 just yesterday. This is a valid issue that is being fixed for NCS 2.3.0.

Author: Rebase gone wrong.

Author: I'm not removing it. I'm just moving it down a couple of lines (see line 1569).

> NCSDK-18108: ``s1`` variant image configuration mismatch

Reviewer 15: NCSDK-18108 is fixed

Author: Yes, the tag update referred to NCSDK-11308. (The tag always precedes the known issue, not follows it.) Is NCSDK-11308 also fixed, @osaether ?

Reviewer 15: @greg-fer Sorry about that. 11308 is not fixed. There is a PR with a fix but it is not marked with NCS 2.3 milestone.

.. note::
    The Secure Partition Manager (SPM) is deprecated as of |NCS| v2.1.0 and removed after |NCS| v2.2.0. It is replaced by :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.

.. rst-class:: v2-3-0 v2-2-0

> NCSDK-19156: Building SPM for other boards than ``nrf5340dk_nrf5340_cpuapp`` and ``nrf9160dk_nrf9160`` fails with compilation error in cortex_m_systick.c

Reviewer 15: SPM was removed in v2.2.0 so this is not an issue in v2.3

Author: Removed tag for NCSDK-19156 and NCSIDB-114, @osaether .

NFC
===

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 14: Fixed for 2.3.0

     cd <NCS_root_directory>
     git checkout v2.1.0 -- subsys/nfc/ndef/ch_record_parser.c

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

Reviewer 14: Fixed for 2.3.0

Multiprotocol Service Layer (MPSL)
==================================

> 

Reviewer 16: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0
 
DRGN-18308: An assert could happen if there is more than one user of the scheduler
   Examples of users of the scheduler include bluetooth, 802.15.4, timeslot (e.g., flash driver).

v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0
DRGN-18555: An assert could happen if requesting timeslots with type MPSL_TIMESLOT_REQ_TYPE_EARLIEST
   When requesting timeslots with type MPSL_TIMESLOT_REQ_TYPE_EARLIEST, an assert indicating that there is already an EARLIEST request pending in MPSL could occur.

  An operation under heavy load can end with a crash, with the MPSL message ``ASSERT: 108, 659``.
  This issue was observed very rarely during stress tests on the devices from the nRF53 and nRF52 Series.

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

Reviewer 17: This've been fixed in 2.3.0.

In addition to the known issues listed here, see also :ref:`softdevice_controller_limitations` for permanent limitations.

> 

Reviewer 16: Some new known issues copied from nrfxlib changelog "bugfix" section (FYI @rugeGerritsen ): 

rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-18261: Wrongly uses non-zero randomness for the first advertising event
   The controller used non-zero randomness even after calling :c:func:`sdc_hci_cmd_vs_set_adv_randomness` with a valid ``adv_handle`` parameter

rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-18358: Scanning may get hardfaults
   The controller would end up in the HardFault handler after receiving an invalid response to a scan request.

rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0

DRGN-18411: LE Connection complete event has wrong peer_address_type if resolved address
   The ``Peer_Address_Type`` parameter in the ``LE Connection Complete`` event was incorrectly set to 2 or 3 in case the connection was established to a device which address was resolved.

rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

DRGN-18420: Periodic advertiser may get NULL pointer dereference
   The controller could dereference a NULL pointer when starting a periodic advertiser.

rst-class:: v2-2-0

DRGN-18586: Assert when starting Periodic Advertisement Sync Transfer while Periodic Advertising is not enabled
   When initiating Periodic Advertising Sync Transfer (PAST) as advertiser, the controller might assert if the periodic advertising train is not running.

rst-class:: v2-2-0

DRGN-18655:  Wrongly set the address if calling :c:func:`bt_ctlr_set_public_addr` before :c:func:`bt_enable
   `bt_ctlr_set_public_addr()` accesses uninitialized memory if called before `bt_enable()`

Author: The 18655 entry is missing description, @laxiLang , and the description is required.

Author: > Some new known issues

@laxiLang , "some" suggests the list is not complete. Are these all known issues that need to be added?

Reviewer 16: If @rugeGerritsen has no further comments, "some" --> whole list (w.r.t posted in nrfxlib sdc changelog)

.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

> DRGN-18568: When the :kconfig:option:`CONFIG_MPSL_FEM` Kconfig option is used, the actual value of radio output power is lower than expected.

Reviewer 16: @rugeGerritsen : DRGN-18568 has been fixed by https://github.com/nrfconnect/sdk-nrf/pull/10143 ?

Reviewer 18: Yes

Reviewer 16: So here,  v2-3-0 should be removed

DRGN-18411: Bluetooth LE Connection complete event has wrong :c:type:`peer_address_type` if resolved address
  The :c:type:`peer_address_type` parameter in the Bluetooth LE Connection Complete event is incorrectly set to ``2`` or ``3`` in case the connection is established to a device whose address was resolved.

.. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

DRGN-18420: Periodic advertiser can get ``NULL`` pointer dereference
  The controller could dereference a ``NULL`` pointer when starting a periodic advertiser.

.. rst-class:: v2-2-0

DRGN-18586: Assert when starting Periodic Advertisement Sync Transfer while Periodic Advertising is not enabled
  When initiating Periodic Advertising Sync Transfer (PAST) as advertiser, the controller might assert if the periodic advertising train is not running.

.. rst-class:: v2-2-0

DRGN-18655: Wrongly set the address if calling :c:func:`bt_ctlr_set_public_addr` before :c:func:`bt_enable`
  :c:func:`bt_ctlr_set_public_addr` accesses uninitialized memory if called before :c:func:`bt_enable`.

.. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0

> DRGN-18568: When the :kconfig:option:`CONFIG_MPSL_FEM` Kconfig option is used, the actual value of radio output power is lower than expected.

Reviewer: ```suggestion
DRGN-18568: When the :kconfig:option:`CONFIG_MPSL_FEM` Kconfig option is used, the actual value of radio output power is lower than expected
```

Author: Rephrased

  **Workaround:** Edit the affected configurations file so that ``align`` is correctly placed inside the ``placement`` section of the config file.

> .. rst-class:: v2-3-0 v2-2-0

Reviewer: NCSDK-15909 - duplicate 'with'

.. rst-class:: v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0

NCSDK-15909: TF-M failing to build with Zephyr SDK 0.14.2
  TF-M may fail to build due to flash overflow with Zephyr SDK 0.14.2 when ``TFM_PROFILE_TYPE_NOT_SET`` is set to ``y``.

  **Workaround:** Use one of the following workarounds:

  * Increase the TF-M partition :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM`.
  * Use Zephyr SDK version 0.14.1.

.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0

NCSDK-16916: TF-M non-secure storage size might be incorrect when having multiple storage partitions

>   TF-M non-secure storage partition ``nonsecure_storage`` size is incorrectly calculated when it has multiple storage partitions inside.

Reviewer 19: NCSDK-16916 affects
v2-3-0 v2-2-0 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0

Author: OK, removing 2.1.3 from the list.

Zephyr
******

> .. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0

Reviewer 20: ```suggestion
.. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0
```

This issue is fixed and fix is https://github.com/nrfconnect/sdk-zephyr/commit/f4c6943be9e4824bea8ded30f5682143ca5df15e

  **Workaround:** Add the text ``zephyr_link_libraries(-Wl,--undefined=my_missing_debug_symbol)`` in the application's :file:`CMakeLists.txt` file.

.. rst-class:: v2-3-0 v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0

> NCSDK-12342: SecureFault exception while accessing protected storage

Reviewer 19: ```suggestion
NCSDK-12342: A SecureFault exception may trigger while accessing protected storage
```

Samples
*******

.. rst-class:: v2-3-0 v2-2-0

NCSDK-18847: :ref:`radio_test` sample does not build with support for Skyworks front-end module
  When building a sample with support for a front-end module different from nRF21540, the sample uses a non-existing configuration to initialize TX power data.
  This causes a compilation error because the source file containing code for a generic front-end module is not included in the build.

  **Workaround:** Do not use the :kconfig:option:`CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC` Kconfig option and replace ``CONFIG_GENERIC_FEM`` with ``CONFIG_MPSL_FEM_SIMPLE_GPIO`` in the :file:`CMakeLists.txt` file of the sample.

.. rst-class:: v2-3-0

NCSDK-19858: :ref:`at_monitor_readme` library and :ref:`nrf_cloud_mqtt_multi_service` sample heap overrun

>   Occasionally, the :ref:`at_monitor_readme` library heap becomes overrun, presumably due to one of the registerred AT event listeners becoming stalled.

Reviewer: ```suggestion
  Occasionally, the :ref:`at_monitor_readme` library heap becomes overrun, presumably due to one of the registered AT event listeners becoming stalled.
```

Samples
*******

> .. rst-class:: v2-3-0 v2-2-0

Reviewer 14: Fixed for 2.3.0.

.. rst-class:: v2-2-0

NCSDK-18847: :ref:`radio_test` sample does not build with support for Skyworks front-end module
  When building a sample with support for a front-end module different from nRF21540, the sample uses a non-existing configuration to initialize TX power data.
  This causes a compilation error because the source file containing code for a generic front-end module is not included in the build.

  **Workaround:** Do not use the :kconfig:option:`CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC` Kconfig option and replace ``CONFIG_GENERIC_FEM`` with ``CONFIG_MPSL_FEM_SIMPLE_GPIO`` in the :file:`CMakeLists.txt` file of the sample.

.. rst-class:: v2-3-0

NCSDK-19858: :ref:`at_monitor_readme` library and :ref:`nrf_cloud_mqtt_multi_service` sample heap overrun
  Occasionally, the :ref:`at_monitor_readme` library heap becomes overrun, presumably due to one of the registered AT event listeners becoming stalled.
  This has only been observed with the :ref:`nrf_cloud_mqtt_multi_service` sample.

Zephyr
******

> 

Reviewer 21: @greg-fer , we need to add information about the following known vulnerabilities:
* https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-3806
* https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-0396
* https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-0397
* https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-0359

Should I create a PR to this PR or you will add them?

  When building a sample with support for a front-end module different from nRF21540, the sample uses a non-existing configuration to initialize TX power data.
  This causes a compilation error because the source file containing code for a generic front-end module is not included in the build.

  **Workaround:** Do not use the :kconfig:option:`CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC` Kconfig option and replace ``CONFIG_GENERIC_FEM`` with ``CONFIG_MPSL_FEM_SIMPLE_GPIO`` in the :file:`CMakeLists.txt` file of the sample.

.. rst-class:: v2-3-0

NCSDK-19858: :ref:`at_monitor_readme` library and :ref:`nrf_cloud_mqtt_multi_service` sample heap overrun
  Occasionally, the :ref:`at_monitor_readme` library heap becomes overrun, presumably due to one of the registered AT event listeners becoming stalled.
  This has only been observed with the :ref:`nrf_cloud_mqtt_multi_service` sample.

Zephyr
******

.. rst-class:: v2-3-0 v2-2-0

NCSDK-18426: Fault when using RTT on nRF52 Series devices in specific circumstances
  Firmware can fault when RTT is enabled on some projects.
  The underlying cause of this issue has not yet been determined.

>   **Workaround:** Disable RTT.

Reviewer 22: This can be removed now, the cause of this issue has been identified and the issue is NCSDK-20104 (details in above comment)

Author: And I see NCSDK-20104 is fixed for 2.3.0?

Reviewer 22: Yes

  **Workaround:** Create a CMake preload script containing ``FOO`` and ``BAR`` settings, and then specify ``-C <pre-load-script>.cmake`` in :guilabel:`Extra CMake Build Options`.

.. _known_issues_vulnerabilities:

Security vulnerabilities
************************

Security vulnerabilities listed in this section *and* tagged with the :ref:`latest official release version <release_notes>` are valid for the current state of development.
Use the drop-down filter at the top of the page to see vulnerabilities for previous releases and check if they are still valid.

> .. note::
>     The list of security vulnerbilities has been introduced only in the |NCS| v2.3.0.
>     For this reason, no previous versions are listed.

Reviewer: ```suggestion
.. note::
   The list of security vulnerabilities has been introduced only in the |NCS| v2.3.0.
   For this reason, no previous versions are listed.
```
